target_snowflake/connector.py

```diff
@@ -0,0 +1,59 @@
+"""Defines the SnowflakeConnector class."""
+
+from typing import Any, Dict
+from singer_sdk import SQLConnector
+
+from snowflake.sqlalchemy import URL, VARIANT, ARRAY, TIMESTAMP_TZ
+import sqlalchemy
+
+
+class SnowflakeConnector(SQLConnector):
+    """The SQLConnector class for Snowflake."""
+
+    def get_sqlalchemy_url(self, config: Dict[str, Any]) -> str:
+        """Concatenate a SQLAlchemy connection URL.format
+
+        Args:
+            config: The tap's config dict.
+
+        Returns:
+            The connection url as a string.
+        """
+        conf = config["snowflake"]
+        return URL(
```

Comment:
Some users may have a harder time creating their own snowflake_url, and there may also be some params we want to inject into that url at runtime. It might make a good option to provide as an override of the piecemeal approach but I think providing them individually may still be helpful for certain cases and users.